include/rocksdb/db.h

```diff
@@ -1587,6 +1587,15 @@ class DB {
     return Status::NotSupported("EndTrace() is not implemented.");
   }
 
+  virtual Status StartIOTrace(Env* /*env*/, const TraceOptions& /*options*/,
```

Comment:
Why not have the method take in the trace file name (rather than the trace writer) and have the trace writer created internally?  I think you have all of the other arguments and it would make for a cleaner interface.

Reply:
Other tracers like BlockCacheTrace and Tracer, all have start trace functions that take TraceWriter as argument. So I added this to make it consistent with other tracers.   So user set trace_writer through  NewFileTraceWriter(Env* env, const EnvOptions& env_options, const std::string& trace_filename, std::unique_ptr<TraceWriter>* trace_writer); and pass that to StartTrace/ StartBlockCacheTrace/ StartIOTrace.